src/app/work-item/work-item-board/work-item-board.component.ts

```diff
@@ -90,6 +90,11 @@ export class WorkItemBoardComponent implements OnInit, OnDestroy {
     private areaService: AreaService,
     private filterService: FilterService,
     private route: ActivatedRoute) {
+      router.events.subscribe(event => {
```

Comment:
I understand, but still no playing with routers anywhere, other than filtering and querying. Instead, we can use workitemService. We can keep a BehaviorSubject in work item service and any component like board can watch over it for update or add item.